tips/TIP-0024/tip-0024.md

```diff
@@ -0,0 +1,178 @@
+            <td>Payload Type</td>
+            <td>uint32</td>
+            <td>
+              The type of the payload. It will instruct the node how to parse the fields that follow.
+            </td>
+          </tr>
+          <tr>
+            <td>Data Fields</td>
+            <td>ANY</td>
+            <td>A sequence of fields, where the structure depends on <code>Payload Type</code>.</td>
+          </tr>
+        </table>
+      </details>
+  <tr>
+    <td>Nonce</td>
+    <td>uint64</td>
+    <td>The nonce which lets this block fulfill the PoW requirement.</td>
+  </tr>
+</table>
+
+## Syntactic validation
+
+The Tangle can only contain syntactically valid blocks. Invalid must be rejected by the node.
+The following criteria defines whether the block passes the syntactic validation:
+
+- The total length of the serialized block must not exceed `Max Block Length`.
+- `Protocol Version` must match the `Protocol Version` config parameter of the node.
+- Parents:
+  - `Parents Count` must be at least 1 and not larger than `Max Parents Count`.
+  - `Parents` must be sorted in lexicographical order.
+  - Each `Block ID` must be unique.
+- Payload (if present):
+  - `Payload Type` must match one of the values described under [Payloads](#payloads).
+  - `Data Fields` must be correctly parsable in the context of the `Payload Type`.
+  - The payload itself must pass syntactic validation.
+- `Nonce` must be valid with respect to the PoW condition described under [Payloads](#payloads). The PoW score itself is computed according to [TIP-12](../TIP-0012/tip-0012.md).
+- There must be no trailing bytes after all block fields have been parsed.
+  
+### PoW validation
+
+The PoW that needs to be performed for each block protects the network against denial-of-service attacks where in a short time too many blocks are issued for the nodes to process. As the processing time of a block heavily depends on the contained payload, the PoW check can also depend on the `Payload Type` and is described under [Payloads](#payloads).
+It is important to note, that the actual parsing and validating of a payload can be computationally expensive. Thus, it is recommended to first parse the block with all its fields including `Payload Type` (but not parsing or validating the actual payload `Data Fields`). Now, simple syntactic validation steps &ndash; including PoW validation &ndash; can be performed and invalid blocks already filtered out before the payload is validated. With this approach, payload-based PoW validation is not significantly more expensive than payload-agnostic validation.
+
+## Payloads
+
+While blocks without a payload, i.e. `Payload Length` set to zero, are valid, such blocks do not contain any information. As such, blocks usually contain a payload. The detailed specification of each payload type is out of scope of this TIP. The following table lists all currently specified payloads that can be part of a block and links to their specification:
```

Comment:
`all` is not completely correct as treasury transaction is currently still part of stardust

Reply:
A treasury transaction may only be put in a milestone receipt if I'm not mistaken.

Reply:
Yeah ignore me, I'm tired and already a bit on holidays 🤦🏻